### PR TITLE
OAK-10796 - Avoid creation of intermediate StringBuilder in JsopBuilder

### PR DESCRIPTION
**Pull request by nfsantos**
Since `invokedynamic` was added to Java, this code in the JsopBuilder class:

```
StringBuilder buff = new StringBuilder(length + 2);
return buff.append('\"').append(s).append('\"').toString(); 
```

can be written more succinctly and efficiently like this:
```
 return '\"' + s + '\"';
```

https://www.baeldung.com/java-string-concatenation-invoke-dynamic

Results of JsopTest micro benchmark with 100M iterations:

Baseline performance:
```
19.02 seconds dummy: 1210065408
18.67 seconds dummy: 1210065408
18.62 seconds dummy: 1210065408
18.65 seconds dummy: 1210065408
18.62 seconds dummy: 1210065408
```
With String concatenation instead of StringBuilder:
```
17.84 seconds dummy: 1210065408
17.68 seconds dummy: 1210065408
17.68 seconds dummy: 1210065408
17.72 seconds dummy: 1210065408
17.69 seconds dummy: 1210065408
```

### CODE REVIEW

File: oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/json/JsopBuilder.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public class JsopBuilder implements JsopWriter {
 
     private static final boolean JSON_NEWLINES = false;
 
-    private StringBuilder buff = new StringBuilder();
+    private final StringBuilder buff = new StringBuilder();
```
Comment (reschke) [inline oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/json/JsopBuilder.java]: what is this change about?

**Review by reschke — CHANGES_REQUESTED** [review]

**Comment by reschke** [comment]
FWIW, if this is merged, by all means use "squash + merge".

File: oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/json/JsopBuilder.java (hunk 1 above)
Comment (nfsantos) [reply oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/json/JsopBuilder.java]: The field is not changed, so it should be final to convey the intention that it should not change and to get the compiler to enforce this invariant. This was causing a warning in IntelliJ and as it was in the same file that I was modifying, I added this small improvement.

**Review by steffenvan — APPROVED** [review]

**Review by fabriziofortino — APPROVED** [review]

File: oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/json/JsopBuilder.java (hunk 1 above)
Comment (reschke) [reply oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/json/JsopBuilder.java]: Please: no. Changes for a ticket should only for the issue desribed in the ticket (or extend the ticket description).

Comment (nfsantos) [reply oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/json/JsopBuilder.java]: reverted.

**Review by reschke — APPROVED** [review]

**Pull request merged**